chain/chain/src/store.rs

```diff
@@ -1995,14 +2177,15 @@ impl<'a> ChainStoreUpdate<'a> {
         for transaction in self.chain_store_cache_update.transactions.iter() {
             store_update.set_ser(ColTransactions, transaction.get_hash().as_ref(), transaction)?;
         }
-        for trie_changes in self.trie_changes.drain(..) {
-            trie_changes
+        for wrapped_trie_changes in self.trie_changes.drain(..) {
+            wrapped_trie_changes
                 .insertions_into(&mut store_update)
                 .map_err(|err| ErrorKind::Other(err.to_string()))?;
-            trie_changes
+            wrapped_trie_changes
                 .key_value_changes_into(&mut store_update)
                 .map_err(|err| ErrorKind::Other(err.to_string()))?;
-            // TODO: save deletions separately for garbage collection.
+            let (block_hash, trie_changes) = wrapped_trie_changes.get_trie_changes();
+            store_update.set_ser(ColTrieChanges, block_hash.as_ref(), &trie_changes)?;
```

Comment:
Following the design of the above `insertions_into`, I would expect a similar helper instead of `get_trie_changes` exposing internals.

Reply:
++